typeagent/aitools/utils.py

```diff
@@ -299,3 +304,8 @@ def make_agent[T](cls: type[T]) -> Agent[None, T]:
         )
 
     return Agent(model, output_type=Wrapper(cls, strict=True), retries=3)
+
+
+def chunk_input(data: list[T] | str, chunk_size: int) -> Iterator[list[T] | str]:
```

Comment:
This would be a good place to use `@overload` to indicate that if the input is a list[T], so is the output, and ditto str->str.